common/primitives/src/signatures.rs

```diff
@@ -270,21 +274,14 @@ fn check_ethereum_signature<L: Lazy<[u8]>>(
 	};
 
 	// signature of ethereum prefixed message eip-191
-	let message_prefixed = eth_message_hash(&format!("0x{:?}", HexDisplay::from(&msg.get())));
+	let message_prefixed = eth_message_hash(&msg.get());
 	if verify_signature(&signature.as_ref(), &message_prefixed, signer) {
 		return true
 	}
 
 	// signature of raw payload, compatible with polkadotJs signatures
 	let hashed = sp_io::hashing::keccak_256(&msg.get());
-	if verify_signature(signature.as_ref(), &hashed, signer) {
-		return true
-	}
-
-	// frequency wrapped for Metamask compatibility
-	let frequency_wrapped =
-		eth_message_hash(&format!("<Frequency>0x{:?}</Frequency>", HexDisplay::from(&msg.get())));
-	verify_signature(&signature.as_ref(), &frequency_wrapped, signer)
```

Comment:
Removing this custom wrapping for now. We might not need it and we will cross that bridge in future in case it was necessary.